src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -416,62 +447,59 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
         }
 
         int numChoicesSelected = 0;
+
         for (FeedbackResponseAttributes response : responses) {
-            List<String> answerStrings =
-                    ((FeedbackMsqResponseDetails) response.getResponseDetails()).getAnswerStrings();
-            boolean isOtherOptionAnswer =
-                    ((FeedbackMsqResponseDetails) response.getResponseDetails()).isOtherOptionAnswer();
-            String otherAnswer = "";
-
-            if (isOtherOptionAnswer) {
-                if (!answerFrequency.containsKey("Other")) {
-                    answerFrequency.put("Other", 0);
-                }
+            numChoicesSelected = numChoicesSelected + getNumOfChoicesSelected(response, answerFrequency);
+        }
+        if (!containsNonEmptyResponse) {
+            return -1;
+        }
 
-                answerFrequency.put("Other", answerFrequency.get("Other") + 1);
+        return numChoicesSelected;
+    }
 
-                numChoicesSelected++;
-                // remove other answer temporarily to calculate stats for other options
-                otherAnswer = answerStrings.get(answerStrings.size() - 1);
-                answerStrings.remove(otherAnswer);
+    private int getNumOfChoicesSelected(FeedbackResponseAttributes response,
```

Comment:
Similar to above. Need some header comment to explain how the method changes answerFrequency.